wire-compiler/src/main/java/com/squareup/wire/WireCompiler.java

```diff
@@ -111,10 +118,14 @@
   final boolean namedFilesOnly;
   final boolean emitAndroid;
   final boolean emitCompact;
+  final ImmutableMap<ProtoType, TypeName> protoTypeToJavaClassName;
+  final ImmutableMap<ProtoType, Map.Entry<ClassName, String>> protoTypeToAdapter;
 
   WireCompiler(FileSystem fs, WireLogger log, List<String> protoPaths, String javaOut,
       List<String> sourceFileNames, IdentifierSet identifierSet, boolean dryRun,
-      boolean namedFilesOnly, boolean emitAndroid, boolean emitCompact) {
+      boolean namedFilesOnly, boolean emitAndroid, boolean emitCompact,
+      Map<ProtoType, TypeName> protoTypeToJavaClassName,
+      Map<ProtoType, Map.Entry<ClassName, String>> protoTypeToAdapter) {
```

Comment:
Are you using `Map.Entry` as a pair? That’s a bit weird. I think it’s probably worth the effort to define a little class `AdapterMember` or something that contains the two fields, ClassName and String.